app/fg/location-bar/index.js

```diff
@@ -290,16 +289,21 @@ class LocationBar extends LitElement {
     var queryId = ++this.queryIdCounter
     this.inputValue = this.inputValue.trim()
     var finalResults
+    var settings = await bg.beakerBrowser.getSettings()
```

Comment:
We dont want to call `getSettings()` here because it gets called multiple times while the user types and needs to be quick. It should happen in `reset()` and the settings can be stored on the `this`

Reply:
Sounds good. Since `reset` is not async and `getSettings` is, I changed it so reset calls `this.settingsPromise = bg.beakerBrowser.getSettings()` and `queryAutocomplete` calls `var settings = await this.settingsPromise`. It works, but I am not sure if there is a better/proper way?